rpc/client.go

```diff
@@ -265,18 +265,19 @@ func (c *Client) heartbeat() error {
 				remoteTimeNow := response.ServerTime + (receiveTime-sendTime)/2
 				c.offset.Offset = remoteTimeNow - receiveTime
 			}
+			offset := c.offset
```

Comment:
They could be a separate commit, but its fairly common to fix associated problems in code that is being edited. In this case, I happened to notice that `c.offset` was being used outside of the lock which is protecting it.